src/core/or/circuit_st.h

```diff
@@ -177,6 +183,18 @@ struct circuit_t {
   /** Hashtable node: used to look up the circuit by its HS token using the HS
       circuitmap. */
   HT_ENTRY(circuit_t) hs_circuitmap_node;
+
```

Comment:
Hmm, I see we are adding these three new padding-related elements to the circuit.
Is it possible to document where and when these get initialized and how they interact with each other?

IIUC, padding_machine comes from the consensus. But can they be NULL? What if the consensus does not specify any padding machines?

Also padding_info specifies the current state of the machines? When is it NULL? When the machine is not yet active?

What about padding_handles? Does it just point to padding_info?

Reply:
821ced37805e9d2ca1110de9ade8d0bfef3aa5e6 maybe clarifies this a bit?

Also I got rid of the handles. They are not needed (because we always cancel timers when freeing padding_info) and just take up space.